sway/commands/input/xkb_file.c

```diff
@@ -0,0 +1,26 @@
+#define _POSIX_C_SOURCE 200809L
+#include "sway/config.h"
+#include "sway/commands.h"
+#include "log.h"
+
+struct cmd_results *input_cmd_xkb_file(int argc, char **argv) {
+	struct cmd_results *error = NULL;
+	if ((error = checkarg(argc, "xkb_file", EXPECTED_EQUAL_TO, 1))) {
+		return error;
+	}
+	struct input_config *ic = config->handler_context.input_config;
+	if (!ic) {
+		return cmd_results_new(CMD_FAILURE, "No input device defined.");
+	}
+
+	if (strcmp(argv[0], "-") == 0) {
```

Comment:
Oops. Sorry didn't think this through. `ic->xkb_file` is always going to be `NULL` here since it's a delta input config so the `free` can be removed. Also, setting it to `NULL` means it won't get merged since there is no difference between not being set at all and being unset. There needs to be a bool or something denoting that this should be merged.

Reply:
I don't understand? I am not too far into the source of sway to understand what is meant by "delta input config?" Heads up, I am heading to bed right now, I will gladly continue working on this in the morning.

Reply:
`config->handler_context.input_config` contains an empty input config. It is used as a delta input config to store the changes that should occur for the input subcommand. The whole storage process is pretty long to express in words. I'd recommend looking at the main input command handler and walking through the `store_input_config` function.

The issue here is that `merge_input_config` will see `src->xkb_file` as `NULL` in both the case that the subcommand does not touch `xkb_file` and the case when it wants to unset the `xkb_file`. When `src->xkb_file` is `NULL`, it doesn't merge it on to the destination input config because it views it as being untouched. There needs to be some way to differentiate between `untouched` and `set it to NULL`.

Adding a simple `bool xkb_file_is_set` to `struct input_config` should work. This could make the section in `merge_input_config` look something like:
```c
if (src->xkb_file_is_set) {
    free(dst->xkb_file);
    dst->xkb_file = src->xkb_file;
    dst->xkb_file_is_set = dst->xkb_file != NULL;
}
```
And then in `store_input_config`:
```c
ic->xkb_file_is_set = ic->xkb_file != NULL;

if (!current || new_current) {  // This line exists, just showing for placement reference
```

Clearing the flag on storage is probably necessary for input identifier configs to get properly merged with input type configs when the input device is originally hotplugged (the type for the identifier is unknown until the device appears).

Hopefully that helps. If you have more questions, let me know.

Reply:
I think I got this; I just pushed the relevant changes.